src/Core/Domain/CartRule/Command/AddCartRuleCommand.php

```diff
@@ -450,89 +353,38 @@ public function setCustomerId(int $customerId): AddCartRuleCommand
         return $this;
     }
 
-    /**
-     * @param bool $hasCountryRestriction
-     *
-     * @return AddCartRuleCommand
-     */
-    public function setHasCountryRestriction(bool $hasCountryRestriction): AddCartRuleCommand
-    {
-        $this->hasCountryRestriction = $hasCountryRestriction;
-
-        return $this;
-    }
-
-    /**
-     * @param bool $hasCarrierRestriction
-     *
-     * @return AddCartRuleCommand
-     */
-    public function setHasCarrierRestriction(bool $hasCarrierRestriction): AddCartRuleCommand
-    {
-        $this->hasCarrierRestriction = $hasCarrierRestriction;
-
-        return $this;
-    }
-
-    /**
-     * @param bool $hasGroupRestriction
-     *
-     * @return AddCartRuleCommand
-     */
-    public function setHasGroupRestriction(bool $hasGroupRestriction): AddCartRuleCommand
-    {
-        $this->hasGroupRestriction = $hasGroupRestriction;
-
-        return $this;
-    }
-
-    /**
-     * @param bool $hasCartRuleRestriction
-     *
-     * @return AddCartRuleCommand
-     */
-    public function setHasCartRuleRestriction(bool $hasCartRuleRestriction): AddCartRuleCommand
-    {
-        $this->hasCartRuleRestriction = $hasCartRuleRestriction;
+    public function setMinimumAmount(
+        string $minimumAmount,
```

Comment:
```suggestion
        DecimalNumber $minimumAmount,
```

We favor using scalar types, but `DecimalNumber` is an exception as we can't rely on float, but string is not really adapted either so we can consider `DecimalNumber` like a scalar type

Reply:
I see we used `string` as input in Product commands, so we better stay consistent with this :smile: